### PR TITLE
Wait longer in tests where namespace cleanup is slow

### PR DESCRIPTION
**Pull request by smarterclayton**
Also fix a case where we waited for dockercfg but not the token (which we need).

This fixes the failure to pull openshift/tools in a number of tests (we raced with the secret controller)

### CODE REVIEW

File: test/extended/util/framework.go [hunk 1]
```diff
@@ -984,13 +984,20 @@ func WaitForServiceAccount(c corev1client.ServiceAccountInterface, name string)
 			return false, fmt.Errorf("Failed to get service account %q: %v", name, err)
 		}
 		secretNames := []string{}
+		var hasDockercfg, hasToken bool
 		for _, s := range sc.Secrets {
-			if strings.Contains(s.Name, "dockercfg") {
-				return true, nil
+			if strings.Contains(s.Name, "-dockercfg-") {
```
Comment (deads2k) [inline test/extended/util/framework.go:989]: This one needs to be in `sc.ImagePullSecrets`.

Comment (smarterclayton) [reply test/extended/util/framework.go]: Good catch

**Pull request merged**